punchcut/src/source/glyf/deltas.rs

```diff
@@ -0,0 +1,289 @@
+            {
+                *working_point = point.map(Fixed::from_i32);
+                flag.clear_marker(PointMarker::HAS_DELTA);
+            }
+            for tuple_delta in tuple.deltas() {
+                let ix = tuple_delta.position as usize;
+                if let (Some(flag), Some(point)) = (flags.get_mut(ix), working_points.get_mut(ix)) {
+                    flag.set_marker(PointMarker::HAS_DELTA);
+                    *point += tuple_delta.apply_scalar(scalar);
+                }
+            }
+            interpolate_deltas(points, flags, contours, &mut working_points[..])
+                .ok_or(ReadError::OutOfBounds)?;
+            for ((delta, point), working_point) in deltas[..actual_len]
+                .iter_mut()
+                .zip(points)
+                .zip(working_points.iter())
+            {
+                *delta += *working_point - point.map(Fixed::from_i32);
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Interpolate points without delta values, similar to the IUP hinting
+/// instruction.
+///
+/// Modeled after the FreeType implementation: <https://github.com/freetype/freetype/blob/bbfcd79eacb4985d4b68783565f4b494aa64516b/src/truetype/ttgxvar.c#L3881>
+fn interpolate_deltas(
+    points: &[Point<i32>],
+    flags: &[PointFlags],
+    contours: &[u16],
+    out_points: &mut [Point<Fixed>],
+) -> Option<()> {
+    if contours.is_empty() {
+        return Some(());
+    }
+    let mut point_ix = 0usize;
+    for &end_point_ix in contours {
+        let end_point_ix = end_point_ix as usize;
+        let first_point_ix = point_ix;
+        // Search for first point that has a delta.
+        while point_ix <= end_point_ix && !flags.get(point_ix)?.has_marker(PointMarker::HAS_DELTA) {
+            point_ix += 1;
+        }
+        if point_ix <= end_point_ix {
+            let first_delta_ix = point_ix;
+            let mut cur_delta_ix = point_ix;
+            point_ix += 1;
+            // Search for next point that has a delta...
+            while point_ix <= end_point_ix {
+                if flags.get(point_ix)?.has_marker(PointMarker::HAS_DELTA) {
+                    // ... and interpolate intermediate points.
+                    interpolate_range(
+                        points,
+                        cur_delta_ix + 1,
+                        point_ix - 1,
+                        cur_delta_ix,
+                        point_ix,
+                        out_points,
+                    );
+                    cur_delta_ix = point_ix;
+                }
+                point_ix += 1;
+            }
+            // If we only have a single delta, shift the contour.
+            if cur_delta_ix == first_delta_ix {
+                shift_range(
+                    points,
+                    first_point_ix,
+                    end_point_ix,
+                    cur_delta_ix,
+                    out_points,
+                )?;
```

Comment:
shift and interpolate with numerous numeric args where some are offset is not the easiest thing to read. I wonder how we could address that? 

Speculatively,

```rust
// early on
let points = PointJiggler { source: points, dest: out_points };

// later
points.delta(cur_delta_idx).apply_to(first_point_ix..end_point_idx)
```

Reply:
Replaced shift and interpolate with these:
```
jiggler.shift(first_point_ix..=end_point_ix, cur_delta_ix)?;

jiggler.interpolate(
    cur_delta_ix + 1..=end_point_ix,
    RefPoints(cur_delta_ix, first_delta_ix),
)?;
```
So they now take range + ref point(s)